src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,94 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int CAPACITY = 16;
+    private int capacity;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        capacity = CAPACITY;
+        table = new Node[capacity];
+    }
 
     @Override
     public void put(K key, V value) {
+        resize();
+        int index = getIndex(key);
+        Node<K, V> current = table[index];
+        Node<K, V> newNode = new Node(key, value);
+        if (current == null) {
+            table[index] = newNode;
+            size++;
+            return;
+        }
 
+        while (current != null) {
+            if (Objects.equals(current.key, key)) {
+                current.value = value;
+                return;
+            }
+            if (current.next == null) {
+                current.next = newNode;
+                size++;
+                return;
+            }
+            current = current.next;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> i = table[index];
+        while (i != null) {
```

Comment:
don't use one-letter naming outside of math context to maintain readability
```suggestion
        while (currentNode != null) {
```